### PR TITLE
fix: add missing env var

### PR DESCRIPTION
**Pull request by davidsauntson**
So I _think_ this will fix the missing `BROWSERSTACK_BUILD_NAME` key issue

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -275,7 +275,8 @@ def define_browserstack_tests() {
       withEnv([
         "BROWSER=${it.browser}",
         "BROWSERSTACK_CONFIGURATION_OPTIONS=${it.config}",
-        'BROWSERSTACK=true'
+        'BROWSERSTACK=true',
+        "BROWSERSTACK_BUILD_NAME=Design system (Jenkins) ${env.BRANCH_NAME}_${getSha()}"
```
Comment (davidrapson) [inline Jenkinsfile:279]: As `DOCKER_TAG=${env.BRANCH_NAME}_${getSha()}` you could arguably do ` "BROWSERSTACK_BUILD_NAME=Design system (Jenkins) ${env.DOCKER_TAG}"`. But this is a pretty short-lived fix so 🤷‍♂️

**Review by davidrapson — APPROVED** [review]
Looks right to me. But worth confirming the syntax works by forcing the browserstack tests to run on this PR.

File: Jenkinsfile (hunk 1 above)
Comment (davidsauntson) [reply Jenkinsfile:279]: forcing a re-run here: https://github.com/citizensadvice/design-system/pull/1239
looks like a goer: https://jenkins.devops.citizensadvice.org.uk/blue/organizations/jenkins/design-system/detail/PR-1239/1/pipeline

Comment (davidsauntson) [reply Jenkinsfile:279]: it failed but only because no components could  load on iPhone IIX: The Last iPhone.

**Pull request merged**